needle-notion-connector/src/utils/notion-utils.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  DatabaseObjectResponse,
+  PageObjectResponse,
+  SearchResponse,
+} from "@notionhq/client/build/src/api-endpoints";
+import { env } from "~/env";
+import type { NotionError, NotionToken } from "~/models/notion-models";
+
+export async function fetchAccessToken(code: string): Promise<NotionToken> {
+  const clientId = env.NOTION_OAUTH_CLIENT_ID;
+  const clientSecret = env.NOTION_OAUTH_CLIENT_SECRET;
+  const oauthUrl = env.NOTION_OAUTH_URL;
+  const redirectUri = new URLSearchParams(oauthUrl).get("redirect_uri");
+
+  const response = await fetch("https://api.notion.com/v1/oauth/token", {
```

Comment:
just curious, does notion library above, provide a function for this endpoint? if so we can use it

Reply:
no it doesn't support the oauth unfortunately, the notion client requires a token already. i also didn't see a reference in the docs.

Reply:
👍🏻